engines/native/resultset.go

```diff
@@ -101,6 +101,11 @@ func (r *resultSet) ExtractFolder(path string, handler engines.FileHandler) erro
 		if _, ok := err.(*os.PathError); ok && first {
 			return engines.ErrResourceNotFound
 		}
+		// If first path is what we're walking and it's not a directory, then we
+		// didn't find folder at the given path.
```

Comment:
Basically, if we did  `filepath.Walk("/home/jonasfj/myfile.txt", func(abspath string, ...` then if `"/home/jonasfj/myfile.txt"` is a file an not a directory, it will call our callback with `abspath == "/home/jonasfj/myfile.txt"`...

So if we try to export a directory artifact, but the path is actually the path to a file instead of a folder, we return that single file.... If `ExtractFolder` is called with the path to a file, it should return `ErrResourceNotFound`... It's a weird go-walk corner case...